dotCMS/src/main/java/com/dotcms/storage/ContentletMetadataAPIImpl.java

```diff
@@ -0,0 +1,301 @@
+
+        this.generateFullMetadata (contentlet, fullBinaryFieldNameSet, fullMetadataMap, fieldMap, alwaysRegenerateMetadata);
+        this.generateBasicMetadata(contentlet, basicBinaryFieldNameSet, fullBinaryFieldNameSet, basicMetadataMap, fieldMap, alwaysRegenerateMetadata);
+
+        return new ContentletMetadata(fullMetadataMap.build(), basicMetadataMap.build());
+    }
+
+    private void generateBasicMetadata(final Contentlet contentlet,
+                                       final Set<String> basicBinaryFieldNameSet,
+                                       final Set<String> fullBinaryFieldNameSet,
+                                       final ImmutableMap.Builder<String, Map<String, Object>> basicMetadataMap,
+                                       final Map<String, Field> fieldMap,
+                                       final boolean alwaysRegenerateMetadata) throws IOException {
+
+        final String storageType        = Config.getStringProperty(DEFAULT_STORAGE_TYPE, StorageType.FILE_SYSTEM.name());
+        Map<String, Object> metadataMap = Collections.emptyMap();
+        final String metadataBucketName = Config.getStringProperty(METADATA_GROUP_NAME, DEFAULT_METADATA_GROUP_NAME);
+        for (final String basicBinaryFieldName : basicBinaryFieldNameSet) {
+
+            final File file           = contentlet.getBinary(basicBinaryFieldName);
+            final String metadataPath = this.getFileName(contentlet, basicBinaryFieldName);
+
+            if (null != file && file.exists() && file.canRead()) {
+
+                // if already included on the full, the file was already generated, just need to add the basic to the cache.
+                final Set<String> metadataFields = this.getMetadataFields(fieldMap.get(basicBinaryFieldName).id());
+
+                if (fullBinaryFieldNameSet.contains(basicBinaryFieldName)) {
+
+                    // if it is included on the full keys, we only have to store the meta in the cache.
+                    metadataMap = this.fileStorageAPI.generateBasicMetaData(file,
+                            metadataKey -> metadataFields.isEmpty() ? true : metadataFields.contains(metadataKey));
+                    CacheLocator.getContentletCache()
+                            .addMetadataMap(contentlet.getInode() + basicBinaryFieldName, metadataMap);
+                } else {
+
+
+                    metadataMap = this.fileStorageAPI.generateMetaData(file,
+                            new GenerateMetaDataConfiguration.Builder()
+                                    .full(false)
+                                    .override(alwaysRegenerateMetadata)
+                                    .store(true)
+                                    .cache(()-> ContentletCache.META_DATA_MAP_KEY + contentlet.getInode() + basicBinaryFieldName)
+                                    .metaDataKeyFilter(metadataKey -> metadataFields.isEmpty() ? true : metadataFields.contains(metadataKey))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)